addons/account/models/account_tax.py

```diff
@@ -972,6 +972,8 @@ def _prepare_taxes_computation(
             if batch['include_base_amount']:
                 for next_batch in ascending_batches[i + 1:]:
                     for next_tax_data in next_batch['taxes']:
+                        if not next_tax_data['is_base_affected']:
+                            continue
```

Comment:
You must reflect the same change in account_tax.js
https://github.com/odoo/odoo/blob/4e10312b14b66df6d9910a714298f5a72d159155/addons/account/static/src/helpers/account_tax.js#L300